setup_machine.sh

```diff
@@ -0,0 +1,156 @@
+# SCRIPT NAME: setup_machine.sh
+
+# DESCRIPTION: This script is called by main.sh. It sets up the web server on 
+# the Deployment Machine/AWS, including installing Docker and Docker Compose on
+# AWS, i.e., sets up EC2 instance for deployment of web app on AWS.
+
+# AUTHOR: Kim Lew
+
+set -ex
+
+ROOT_DIR='/home/ubuntu'
+PROJECT_DIR='rails_see_stuff'
+
+check_current_directory () {
+  if [ ! -d "${THE_DIR}" ]; then
+    cd "${THE_DIR}"
+  fi
+}
+
+check_existing_project_directory () {
+  if [ -d "${ROOT_DIR}"/"${PROJECT_DIR}" ]; then
+    echo "A previous " "${PROJECT_DIR}" "exists. Removing it for a clean start state."
+    echo 'The directory path is: ' "${ROOT_DIR}"/"${PROJECT_DIR}"
+    rm -r "${ROOT_DIR:?}"/"${PROJECT_DIR}"
+  fi
+}
+
+# Check if nano already installed & install if not. "" or check exit code?
+check_if_nano_installed () {
+  if ! command -v nano &> /dev/null; then
+    sudo apt install nano -y
+  fi
+}
+
+echo
+echo "INSTALLING required packages plus Docker & Docker Compose ON VM on AWS..."
+check_current_directory "${ROOT_DIR}"
+check_existing_project_directory
+
+sudo apt update && sudo apt upgrade -y
+check_if_nano_installed
+
+# --- Docker Installation ---
+# Refer to: https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-22-04
+
+# Install a few prerequisite packages which let apt use packages over HTTPS.
+sudo apt install apt-transport-https ca-certificates curl software-properties-common -y
+
+echo "CHECK curl verion..."
+curl --version
+echo
+# Add the GPG key for the official Docker repository to your system.
+# --batch - to disallow interactive commands
+# -f - 
+# -s - silent
+# -L 
+# -S - Show error
+
+# curl -fsLS https://download.docker.com/linux/ubuntu/gpg | sudo gpg --batch --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+
+# https://docs.docker.com/engine/install/ubuntu/
+# --batch - to get rid of error: gpg: cannot open '/dev/tty': No such device or address
+echo "ADD official GPG key of Docker..."
+sudo mkdir -p /etc/apt/keyrings
+# curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --batch --dearmor -o /etc/apt/keyrings/docker.gpg
+# curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --batch --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+# echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+# sudo chmod a+r /usr/share/keyrings/docker-archive-keyring.gpg
+
+# sudo mkdir -p /usr/share/keyrings
+# GPG_KEY='/usr/share/keyrings/docker-archive-keyring.gpg'
+
+sudo mkdir -p /etc/apt/keyrings
+GPG_KEY='/etc/apt/keyrings/docker.gpg'
+
+# CHECK if gpg key appears to be correct.
+# curl -fsSL https://download.docker.com/linux/ubuntu/gpg > "${GPG_KEY}"
+# echo "SHOW key..."
+# cat "${GPG_KEY}"
+# echo
+# sudo gpg --batch --dearmor -o "${GPG_KEY}"
```

Comment:
There is a lot of old commented-out code here, which is just more stuff to read before you get to the code that actually runs. I suggest removing that and preserving only those comments that are relevant to what the script actually does right now.

Reply:
Deleted irrelevant comments.

Reply:
Confirmed you deleted the old comments.